lib/matplotlib/lines.py

```diff
@@ -759,7 +780,7 @@ def draw(self, renderer):
             x0, x1 = self.axes.get_xbound()
             i0 = self._x_filled.searchsorted(x0, 'left')
             i1 = self._x_filled.searchsorted(x1, 'right')
-            subslice = slice(max(i0 - 1, 0), i1 + 1)
+            subslice = slice(max(i0 - 2, 0), i1 + 2)
```

Comment:
What is this change? Was this wrong before?

Reply:
Thought, I needed it to show edges right. Turns out I had an untested behaviour, so I added a test for low, mid, high lims for edges